rasa/nlu/selectors/embedding_response_selector.py

```diff
@@ -0,0 +1,480 @@
+        self._encoded_all_label_ids = self._create_encoded_labels(
+            label_dict,
+            training_data,
+            attribute="response",
+            attribute_feats="response_features",
+        )
+
+        # check if number of negatives is less than number of intents
+        logger.debug(
+            "Check if num_neg {} is smaller than "
+            "number of intents {}, "
+            "else set num_neg to the number of intents - 1"
+            "".format(self.num_neg, self._encoded_all_label_ids.shape[0])
+        )
+
+        # noinspection PyAttributeOutsideInit
+        self.num_neg = min(self.num_neg, self._encoded_all_label_ids.shape[0] - 1)
+
+        session_data = self._create_session_data(
+            training_data, label_dict, attribute="response"
+        )
+
+        if self.evaluate_on_num_examples:
+            session_data, eval_session_data = train_utils.train_val_split(
+                session_data, self.evaluate_on_num_examples, self.random_seed
+            )
+        else:
+            eval_session_data = None
+
+        # if self.share_embedding:
+        #     if X[0].shape[-1] != Y[0].shape[-1]:
+        #         raise ValueError("If embeddings are shared "
+        #                          "text features and intent features "
+        #                          "must coincide")
+
+        self.graph = tf.Graph()
+        with self.graph.as_default():
+
+            # set random seed
+            tf.set_random_seed(self.random_seed)
+
+            # allows increasing batch size
+            batch_size_in = tf.placeholder(tf.int64)
+
+            (
+                self._iterator,
+                train_init_op,
+                eval_init_op,
+            ) = train_utils.create_iterator_init_datasets(
+                session_data, eval_session_data, batch_size_in, self.batch_strategy
+            )
+
+            self._is_training = tf.placeholder_with_default(False, shape=())
+
+            loss, acc = self._build_tf_train_graph()
+
+            # define which optimizer to use
+            self._train_op = tf.train.AdamOptimizer().minimize(loss)
+
+            # train tensorflow graph
+            self.session = tf.Session()
+            train_utils.train_tf_dataset(
+                train_init_op,
+                eval_init_op,
+                batch_size_in,
+                loss,
+                acc,
+                self._train_op,
+                self.session,
+                self._is_training,
+                self.epochs,
+                self.batch_size,
+                self.evaluate_on_num_examples,
+                self.evaluate_every_num_epochs,
+            )
+
+            # rebuild the graph for prediction
+            self.pred_confidence = self._build_tf_pred_graph(session_data)
+
+    def persist(self, file_name: Text, model_dir: Text) -> Dict[Text, Any]:
+        """Persist this model into the passed directory.
```

Comment:
We use Google's docstring format. You can set it up in pycharm. (tools - python integrated tools)